examples/tree_1d_dgsem_spline_approx/elixir_shallowwater_wall_b_spline.jl

```diff
@@ -0,0 +1,102 @@
+#########################################################################################
+# This example is equivalent to tree_1d_dgsem/elixir_shallowwater_wall.jl,              #
+# but instead of a function for the bottom topography, this version uses a cubic        #
+# b spline interpolation with not-a-knot boundary condition to approximate the bottom   #
+# topography. Interpolation points are provided by a .txt file called data_swe_wall.txt #
+#########################################################################################
+
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# Semidiscretization of the shallow water equations
+
+equations = ShallowWaterEquations1D(gravity_constant=9.812, H0=2.0)
+  
+# bottom bottom_topography function
+# This has been used to create the file "data_swe_wall" with 100 equally spaced data points 
+# in the x-direction
+bottom_topography(x) = (1.5 / exp( 0.5 * ((x - 1.0)^2) )+ 0.75 / exp(0.5 * ((x + 1.0)^2)))
+
+# Spline Interpolation
+spline          = cubic_b_spline(joinpath(examples_dir(), "tree_1d_dgsem_spline_approx", "data_swe_wall2.txt"); boundary = "not-a-knot")
```

Comment:
```suggestion
spline_data = joinpath(examples_dir(), "tree_1d_dgsem_spline_approx", "data_swe_wall2.txt")
spline          = cubic_b_spline(spline_data; boundary = "not-a-knot")
```
Just to make it more obvious that the spline points are coming from a particular file.

Reply:
Updated

Reply:
Has been considered, as I am now using gist